core/src/main/java/io/grpc/internal/AbstractManagedChannelImplBuilder.java

```diff
@@ -379,6 +385,74 @@ public T proxyDetector(@Nullable ProxyDetector proxyDetector) {
     return thisT();
   }
 
+  @Override
+  public T defaultServiceConfig(@Nullable Map<String, ?> serviceConfig) {
+    defaultServiceConfig = parseMap(serviceConfig);
+    return thisT();
+  }
+
+  @Nullable
+  private static Map<String, ?> parseMap(@Nullable Map<?, ?> map) {
```

Comment:
Note: if you instead had `parseObject`, this would be much shorter as it would have less duplication between map and list. But it seems this will be deleted almost immediately, so it is fine as-is.